### PR TITLE
SessionCache documentation enhanced; config keys moved to Constants class

### PR DESCRIPTION
**Pull request by mtrna**
continuation of https://github.com/azkaban/azkaban/pull/1608:

documentation of `SessionCache` enhanced; session TTL config key moved to `Constants`

@HappyRay

### CODE REVIEW

**Comment by HappyRay** [comment]
Could you add to your PR description how you tested this change?

Otherwise LGTM. Thanks!

**Comment by mtrna** [comment]
New: I have added test for `SessionCache` hits/misses. Rebased on the latest master. Fixed some more doc/typos found in the process.

File: azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.server.session;
+
+import azkaban.user.User;
+import azkaban.utils.Props;
+import azkaban.utils.PropsUtils;
+import org.junit.Assert;
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java]: Please use AssertJ for new unit tests.
see https://github.com/azkaban/azkaban/blob/master/CONTRIBUTING.md
>  Use AssertJ as the assertion library in unit tests.

**Comment by HappyRay** [comment]
Thanks for adding unit tests!

Left one comment. Otherwise LGTM.

File: azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java (hunk 1 above)
Comment (mtrna) [reply azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java]: updated

File: azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.server.session;
+
+import azkaban.user.User;
+import azkaban.utils.Props;
+import azkaban.utils.PropsUtils;
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java]: The import organization doesn't seem to be correct.

Did you set up the style and SaveAction plugin as described in the contributor guide?

**Comment by HappyRay** [comment]
Thanks for the update.

Could you submit new commits with a description about each commit? This will help reviewers see the changes better and allow the whole change history to be preserved. 

We will rebase for you when the PR is merged. 

Make sure you keep the PR description updated. We will use that as the final commit message.

File: azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java (hunk 2 above)
Comment (mtrna) [reply azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java]: Thanks for the feedback. I have changed the import order.

Comment (HappyRay) [reply azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java]: Thanks.

Did the SaveAction plugin and style config work for you?

File: azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
```
[the remark this answers is not in the document]
Comment (mtrna) [reply azkaban-common/src/test/java/azkaban/server/session/SessionCacheTest.java]: fixed

**Comment by HappyRay** [comment]
@mtrna sorry about the delay in getting back to you. I have been very busy with other tasks.

Your change LGTM. I will try to merge it now.

I am curious why you decided to close this PR.

**Comment by HappyRay** [comment]
Unfortunately, there is a merge conflict now.

@mtrna are you interested in resolving the conflict? I will try to merge it as soon as you update this PR.

**Comment by HappyRay** [comment]
cc @ameyamk

**Comment by mtrna** [comment]
Hi @HappyRay @ameyamk ! No problem about the delay. I have re-based the MR. It should be conflict free now. ;)

**Pull request merged**